source/blog/2018-03-06-update-kernel-arguments-on-atomic-host.html.md

```diff
@@ -0,0 +1,186 @@
+
+[root@localhost ~]# rpm-ostree ex kargs --replace no_timer_check="" --replace net.ifnames=0
+Copying /etc changes: 21 modified, 0 removed, 92 added
+Transaction complete; bootconfig swap: yes deployment count change: 0
+Kernel arguments updated.
+Run "systemctl reboot" to start a reboot
+[root@localhost ~]# rpm-ostree ex kargs
+The kernel arguments are:
+no_timer_check console=tty1 console=ttyS0,115200n8 net.ifnames=0 biosdevname=0 rd.lvm.lv=atomicos/root root=/dev/mapper/atomicos-root ostree=/ostree/boot.0/fedora-atomic/9a9b350be75846811cbb0b1fd7b3d42a49908ed1265bc59e292bb4a34674332c/0
+```
+
+## 3: Delete one or multiple arguments
+
+To delete one or more kernel arguments, the input will be in the form of key=value format. Similarly, when
+there is only one single key=value pair, we allow users to delete argument by key name.
+
+```
+[root@localhost ~]# rpm-ostree ex kargs --delete no_timer_check --delete console=tty1
+Copying /etc changes: 21 modified, 0 removed, 92 added
+Transaction complete; bootconfig swap: yes deployment count change: 0
+Kernel arguments updated.
+Run "systemctl reboot" to start a reboot
+[root@localhost ~]# rpm-ostree ex kargs
+The kernel arguments are:
+console=ttyS0,115200n8 net.ifnames=0 biosdevname=0 rd.lvm.lv=atomicos/root root=/dev/mapper/atomicos-root ostree=/ostree/boot.1/fedora-atomic/9a9b350be75846811cbb0b1fd7b3d42a49908ed1265bc59e292bb4a34674332c/0
+
+```
+
+## 4: Allow user to do all above actions using an editor
+
+Note: the kernel arguments were from the previous section, and the example below will show
+you the already changed ones because it is hard to demo from a separate editor side.
+
+```
+[root@localhost ~]# rpm-ostree ex kargs --editor
+
+======================================EDITOR SCREEN=================================================
+# Please enter the kernel arguments. Each kernel argument# should be in the form of key=value.
+# Lines starting with '#' will be ignored. Each key=value pair should be
+# separated by spaces, and multiple value associated with one key is allowed.
+# Also, please note that any changes to the ostree argument will not be
+# effective as they are usually regenerated when bootconfig changes.
+test=test console=ttyS0,115200n8 biosdevname=0 rd.lvm.lv=atomicos/root root=/dev/mapper/atomicos-root
+=====================================END OF EDITOR SCREEN===========================================
+Copying /etc changes: 21 modified, 0 removed, 92 added
+Transaction complete; bootconfig swap: yes deployment count change: 0
+Kernel arguments updated.
+Run "systemctl reboot" to start a reboot
+
+[root@localhost ~]# rpm-ostree ex kargs
+The kernel arguments are:
+test=test console=ttyS0,115200n8 biosdevname=0 rd.lvm.lv=atomicos/root root=/dev/mapper/atomicos-root ostree=/ostree/boot.0/fedora-atomic/9a9b350be75846811cbb0b1fd7b3d42a49908ed1265bc59e292bb4a34674332c/0
+```
+
```

Comment:
Recommendation: Add a conclusion section letting people know where to post feedback and encouraging people to play around with the functionality.

Reply:
Thoughts: I am thinking of adding this sentence to the conclusion, but I am unsure if that is appropriate yet.

The sentence looks like the following: `we always believe customer satisfaction is the key to success, so please let us know if improvement is needed`. That seemed nice, but sounds a bit vague WDYT?